### ISSUE TITLE
AMD modules in NPM

### ISSUE THREAD

**Issue by SiAdcock** [Jameskmonger/courtroom#30]
In the last publish, the Typescript has been compiled down to AMD modules. This is not great for Node, which only supports the CommonJS module strategy. I do like the change to the directory structure, though! 

Can we publish CommonJS modules to NPM? 

```
$ tsc --module commonjs
```

**Comment by SiAdcock** [comment]
To ensure consistency in how we develop, test and publish, we should add build scripts to the package.json:

```json
"scripts": {
  "test": "gulp test",
  "build": "tsc --module amd",
  "build:commonjs": "tsc --module commonjs",
  }
```

The above can be adjusted as necessary, and executed from the command line:

```
$ npm run build
$ npm run build:commonjs
```

**Comment by Jameskmonger** [comment]
What about using UMD rather than AMD or CommonJS?

**Comment by Jameskmonger** [comment]
Fixed in 16f5f286092aba88e92beb59bd57df6699a83324 - we now use UMD to publish to NPM.

**Comment by SiAdcock** [comment]
Cool, UMD works
